### PR TITLE
Set up clang formatting

### LINKED ISSUE #34: Set up clang-format
For consistency with nextSIM-DG.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #34.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,131 +1,163 @@
...
+                        GNUInstallDirs)
+#Define RPATH for executables via a relative expression to enable a
+#fully relocatable package
+                        file(RELATIVE_PATH relDir1 ${CMAKE_CURRENT_BINARY_DIR} /
+                             ${CMAKE_INSTALL_BINDIR} ${
+                                 CMAKE_CURRENT_BINARY_DIR} /
+                             ${CMAKE_INSTALL_LIBDIR})
+                            file(
+                                RELATIVE_PATH relDir2 ${
+                                    CMAKE_CURRENT_BINARY_DIR} /
```
Comment (TomMelt) [inline CMakeLists.txt:30]: I think something has gone wrong with the formatting here :/

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,166 @@
+---
```
Comment (TomMelt) [inline .clang-format:1]: thanks for adding `.clang-format`

File: LICENSE [hunk 3]
```diff
@@ -77,8 +77,10 @@
       use, offer to sell, sell, import, and otherwise transfer the Work,
       where such license applies only to those patent claims licensable
       by such Contributor that are necessarily infringed by their
-      Contribution(s) alone or by combination of their Contribution(s)
-      with the Work to which such Contribution(s) was submitted. If You
+      Contribution(s)
```
Comment (TomMelt) [inline LICENSE:80]: Should the LICENSE be formatted?

File: README.md [hunk 4]
```diff
@@ -181,32 +183,34 @@ To remove disambiguity we renamed the outputs produced in the `partition_metadat
...
+    │  │                   │ top_neighbor_halos = 12, 12;
+    │  │                   │ bottom_neighbors = 1, 1, 0;
+    │  │ 2         │ bottom_neighbor_ids = 2, 2;
+    │  │                   │ bottom_neighbor_halos = 12, 12;
+    │  │                   │ left_neighbors = 0, 1, 0;
+    30 ▼  └───────────────────┘ left_neighbor_ids = 0;
+    left_neighbor_halos = 20;
+    x right_neighbors = 1, 0, 0;
+    right_neighbor_ids = 1;
+    right_neighbor_halos = 20;
```
Comment (TomMelt) [inline README.md:213]: I don't think we should format the README with clang-format either.

If we do though, we need to wrap my beautiful ascii art with the following,

```
// clang-format off
...
// clang-format on
```

**Review by TomMelt — CHANGES_REQUESTED** [review]
Thanks for adding this @jwallwork23 , it will be really useful.

Just a couple of requests

File: README.md (hunk 4 above)
Comment (jwallwork23) [reply README.md]: Oops! Fixed with force-push.

File: LICENSE (hunk 3 above)
Comment (jwallwork23) [reply LICENSE]: Oops! Fixed with force-push.

File: CMakeLists.txt (hunk 1 above)
Comment (jwallwork23) [reply CMakeLists.txt]: Oops! Fixed with force-push.

**Review by TomMelt — APPROVED** [review]
Thanks looks good to me now.

**Pull request merged**